### DISCUSSION TITLE
Keyboard configuration xfreerdp3-x11 : multimon, us-int keyboard euro on 5 and dutch language

### DISCUSSION THREAD

**Discussion by RneeJ** [FreeRDP/FreeRDP#10673] [Show and tell]
Well , the wikipage about keyboard is from 2011. Pointed to domains and web pages that don't exist anymore.
For years i had a fine working xfreerdp configuration to our windows terminal server. I'm the only one with Linux... 

Then came xfreerdp3-x11 and it didn't work any more. The great developers at Freerdp decided it was time for a change.
Well it took me a while to figure out what all was changed, and i am very proud of solved  my configuration.

On Ubuntu 24.04.01 LTS with Wayland.

Well this works fine for me now full 2 screens (4K) , us-int keyboard and euro on 5 :
`
xfreerdp3 /u:<user> /p:<password> /v:<host> /scale:180 /scale-desktop:250 /multimon /kbd:layout:0x00020409,lang:0x00000413
`
Kinda proud of it ! 

Hopes it helps others.

**Comment by akallabeth** [comment]
@RneeJ sorry that it was hard to find, but `xfreerdp /help` lists the command line options. (might differ from distribution to distribution as only compiled in arguments are listed)

These feature the `/list` option (always compiled in) which has the option to query everything keyboard related:
`xfreerdp /list:kbd`, `xfreerdp /list:kbd-scancode` and `xfreerdp /list:kbd-lang`

**Reply by RneeJ** [reply]
Thank you for your quick response. 

I am still a gratefull user. I still prefer the commandline over the gui alternatives, it seems these are missing options, so freerdp just klick and always run the same way. 

But a lot was changed since my previous install (/f gives problems, kbd was before not with a , (comma)seperator, scaling different) For a non freerdp specialist this makes things very difficult when there is a startup no serverwindow nor errors.

Your additions are realy helpfull. 

Mayby a tip for setting up a starters page?

But it works now and please keep up the good work!

**Reply by akallabeth** [reply]
@RneeJ ok, maybe open an issue if you have issues with `/f` (that did not change actually)
as for the `/kbd` stuff, we did keep compatibility but started grouping the options (the command line grew to a monster)
We maybe did not communicate this strong enough, so most distros did not compile with backwards compatibility enabled ( `-DWITH_FREERDP_DEPRECATED_COMMANDLINE=ON` option)
as for documentation pages, ... help is always welcome, but already busy with the code (cleanups, ...)
